app/src/main/java/com/android/calendar/event/EditEventView.java

```diff
@@ -1485,12 +1485,16 @@ public void onItemSelected(AdapterView<?> parent, View view, int position, long
         mModel.mCalendarAccountName = c.getString(EditEventHelper.CALENDARS_INDEX_ACCOUNT_NAME);
         mModel.mCalendarAccountType = c.getString(EditEventHelper.CALENDARS_INDEX_ACCOUNT_TYPE);
         // TODO: try to find a similar color within the new event colors before falling back to the calendar color
-        mModel.setEventColor(Arrays.stream(mModel.getCalendarEventColors())
-                .filter(color -> color == mModel.getEventColor())
-                .findFirst()
-                .orElse(mModel.getCalendarColor()));
+        if (mModel.getCalendarEventColors() != null) {
+            mModel.setEventColor(Arrays.stream(mModel.getCalendarEventColors())
+                    .filter(color -> color == mModel.getEventColor())
+                    .findFirst()
+                    .orElse(mModel.getCalendarColor()));
+            setColorPickerButtonStates(mModel.getCalendarEventColors());
```

Comment:
move this out of the if-block so the color picker gets disabled for calendars without colors.

Reply:
Wasn't the problem here that getCalendarEventColors() can return null? So it wouldn't make much sense to move it

Reply:
`setColorPickerButtonStates()` checks for null and disables the color picker if necassary

```java
public void setColorPickerButtonStates(int[] colorArray) {
        setColorPickerButtonStates(colorArray != null && colorArray.length > 0);
}
```

Reply:
Done